Sources/Sentry/SentryANRTrackingIntegration.m

```diff
@@ -0,0 +1,111 @@
+#import "SentryANRTrackingIntegration.h"
+#import "SentryANRTracker.h"
+#import "SentryCrashAdapter.h"
+#import "SentryDefaultCurrentDateProvider.h"
+#import "SentryDispatchQueueWrapper.h"
+#import "SentryLog.h"
+#import "SentryThreadWrapper.h"
+#import <Foundation/Foundation.h>
+#import <SentryAppState.h>
+#import <SentryAppStateManager.h>
+#import <SentryCrashAdapter.h>
+#import <SentryDependencyContainer.h>
+#import <SentryOptions+Private.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface
+SentryANRTrackingIntegration ()
+
+@property (nonatomic, strong) SentryANRTracker *tracker;
+@property (nonatomic, strong) SentryAppStateManager *appStateManager;
+@property (nullable, nonatomic, copy) NSString *testConfigurationFilePath;
+
+@end
+
+@implementation SentryANRTrackingIntegration
+
+- (instancetype)init
+{
+    if (self = [super init]) {
+        self.testConfigurationFilePath
+            = NSProcessInfo.processInfo.environment[@"XCTestConfigurationFilePath"];
+    }
+    return self;
+}
+
+- (void)installWithOptions:(SentryOptions *)options
+{
+    if ([self shouldBeDisabled:options]) {
+        [options removeEnabledIntegration:NSStringFromClass([self class])];
+        return;
+    }
+
+    SentryDependencyContainer *dependencies = [SentryDependencyContainer sharedInstance];
+    self.appStateManager = dependencies.appStateManager;
+
+    self.tracker =
+        [[SentryANRTracker alloc] initWithDelegate:self
+                             timeoutIntervalMillis:options.anrTimeoutIntervalMillis
+                               currentDateProvider:[SentryDefaultCurrentDateProvider sharedInstance]
+                                      crashAdapter:dependencies.crashAdapter
+                              dispatchQueueWrapper:[[SentryDispatchQueueWrapper alloc] init]
+                                     threadWrapper:dependencies.threadWrapper];
+    [self.tracker start];
+}
+
+- (BOOL)shouldBeDisabled:(SentryOptions *)options
+{
+    if (!options.enableANRTracking) {
+        return YES;
+    }
+
+    SentryCrashAdapter *crashAdapter = [SentryCrashAdapter sharedInstance];
+    if ([crashAdapter isBeingTraced] && !options.enableANRTrackingInDebug) {
+        return YES;
+    }
+
+    // The testConfigurationFilePath is not nil when running unit tests. This doesn't work for UI
+    // tests though.
+    if (self.testConfigurationFilePath) {
+        [SentryLog logWithMessage:@"Won't track ANRs, because detected that unit tests are running."
+                         andLevel:kSentryLevelDebug];
+        return YES;
+    }
```

Comment:
What do you think of using the `#if TEST` directive?

```suggestion
#if TEST
        [SentryLog logWithMessage:@"Won't track ANRs, because detected that unit tests are running."
                         andLevel:kSentryLevelDebug];
        return YES;
#endif
```